app/db/crud.py

```diff
@@ -1,55 +1,135 @@
 """ CRUD operations for management of resources
 """
+import logging
+import datetime
+from typing import List
+from sqlalchemy import or_
 from sqlalchemy.orm import Session
-from db.models import ForecastModel, SingleForecast
-import schemas
+from sqlalchemy.sql.expression import func
+from app.db.models import (
+    ProcessedModelRunFile, PredictionModel, PredictionModelRun, PredictionModelGridSubset, ModelPrediction)
 
 
-def get_forecast_models(db_session: Session):
-    """ Get all available info about all supported weather forecast models """
-    return db_session.query(ForecastModel).all()
+logger = logging.getLogger(__name__)
 
 
-def create_single_forecast(db_session: Session,
-                           forecast: schemas.WeatherForecastValues,
-                           issue_date: str,
-                           weather_station: int):
-    """ Create a new single forecast
-        `weather_station` corresponds to schemas.WeatherStation.code
+LATLON_15X_15 = 'latlon.15x.15'
+
+
+def get_most_recent_model_run(session: Session, abbreviation: str, projection: str) -> PredictionModelRun:
+    """
+    Get the most recent model run of a specified type. (.e.g. give me the global
+    model at 15km resolution)
+
+    params:
+    :abbreviation: e.g. GDPS or RDPS
+    :projection: e.g. latlon.15x.15
+    """
+    # NOTE: There must be a better way to do this! Doing two seperate queries is horrible!
+    most_recent_timestamp = session.query(func.max(PredictionModelRun.prediction_run_timestamp)).\
+        join(PredictionModel).\
+        filter(PredictionModel.abbreviation == abbreviation, PredictionModel.projection == projection).\
+        scalar()
+    return session.query(PredictionModelRun).\
+        filter(PredictionModelRun.prediction_run_timestamp == most_recent_timestamp).\
+        join(PredictionModel).\
+        filter(PredictionModel.abbreviation == abbreviation, PredictionModel.projection == projection).\
+        first()
```

Comment:
We can order PredictionModelRun by timestamp in descending and get the first one no?

Reply:
yes - that's what I'm doing with noon_forecasts

Reply:
Oh duh - yes.